library/lua/gui.lua

```diff
@@ -537,8 +537,11 @@ function Screen:show(parent)
     if self._native then
         error("This screen is already on display")
     end
+    self:onAboutToShow(parent or dfhack.gui.getCurViewscreen(true))
```

Comment:
I had originally written it so that `onAboutToShow()` can optionally return a new parent, but I saw that as too big of a risk. It would be too easy for `onAboutToShow` overrides to forget to do this, or worse, have a superclass method that changes the parent, and the subclass override that calls the superclass implementation but forgets to propagate the new parent. In the end, I figured that if a client knows what they're doing and passes a `parent` to `onShow`, then use that. otherwise autocorrect whenever possible.

Reply:
I can document this in Lua API.rst